src/clr-angular/utils/popover/popover-content.ts

```diff
@@ -98,25 +111,48 @@ export class ClrPopoverContent implements AfterContentChecked, OnDestroy {
     // Create the view container
     this.view = this.container.createEmbeddedView(this.template);
     this.smartEventsService.contentRef = this.view.rootNodes[0]; // So we know where/what to set close focus on
-    // Position the content and add a click listener
     this.renderer.addClass(this.view.rootNodes[0], 'clr-popover-content');
+    // Reset to the begining of the document to be available for sizing/positioning calculations.
+    // If we add new content to the bottom it triggers changes in the layout that may lead to false anchor
+    // coordinates values.
+    this.renderer.setStyle(this.view.rootNodes[0], 'top', '0px');
+    this.renderer.setStyle(this.view.rootNodes[0], 'left', '0px');
+    // We need to hide it during the calculation phase, while it's not yet finally positioned.
+    this.renderer.setStyle(this.view.rootNodes[0], 'opacity', '0');
     this.renderer.listen(this.view.rootNodes[0], 'click', event => {
       this.smartOpenService.openEvent = event;
     });
     this.view.rootNodes.forEach(node => {
       this.renderer.appendChild(this.document.body, node);
     });
+    // Mark for realingment on the next content-check cycle.
+    this.shouldRealign = true;
   }
-  private hasPositionCoords = false;
+
+  private shouldRealign = false;
+
+  // Check-collector pattern:
+  // In order to get accurate content height/width values, we cannot calculate alignment offsets until
+  // after the projected content has stabilized.
+  // As multiple check events may happen in the same rendering cycle, we need to collect all events
+  // and only act after the content is really stable. Or we may get wrong intermediate positioning values.
+  // We will channel subsequent content check events through this observable.
+  private checkCollector: EventEmitter<void> = new EventEmitter();
 
   ngAfterContentChecked(): void {
-    // In order to get accurate content height/width values, we cannot calculate alignment offsets until after the
-    // projected content has stabilized.
-    if (this.smartOpenService.open && this.view && !this.hasPositionCoords) {
-      const positionCoords = this.smartPositionService.alignContent(this.view.rootNodes[0]);
-      this.renderer.setStyle(this.view.rootNodes[0], 'top', `${positionCoords.yOffset}px`);
-      this.renderer.setStyle(this.view.rootNodes[0], 'left', `${positionCoords.xOffset}px`);
-      this.hasPositionCoords = true;
+    if (this.smartOpenService.open && this.view && this.shouldRealign) {
+      // Channel content-check event through the check-collector
+      this.checkCollector.emit();
+    }
+  }
+
+  private doAlign() {
```

Comment:
nit: rename this method to `realignContent` or the like.

Reply:
changed to `alignContent` as it is also what we run on init.